cpp/recognition/cograph/BretscherCorneilHabibPaulCographRecognition.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * BretscherCorneilHabibPaulCographRecognition.cpp
+ *
+ *  Created on: 2024
+ *      Author: fixikmila
+ */
+// Copyright 2024 milana
+#include <graph/GraphTools.hpp>
+
+#include "recognition/CographRecognition.hpp"
+#include "recognition/CoTree.hpp"
+
+#include <list>
+
+namespace Koala {
+
+    bool BretscherCorneilHabibPaulCographRecognition::isCograph() const {
+        assureFinished();
+        return is_cograph;
+    }
+
+    void BretscherCorneilHabibPaulCographRecognition::run() {
+        hasRun = true;
+        std::vector<NetworKit::node> start;
+        for (auto u : graph.nodeRange()) {
+            start.push_back(u);
+        }
+        auto [bb, a] = LexBfsMinus(false, start);
+        auto [borders1, b] = LexBfsMinus(true, a);  // graph complement
+        auto [borders2, c] = LexBfsMinus(false, b);
+        if (NeighbourhoodSubsetProperty(true, b, borders1) &&
+        NeighbourhoodSubsetProperty(false, c, borders2)) {
+            is_cograph = true;
+        } else {
+            is_cograph = false;
+        }
+    }
+
+    std::pair<std::vector<std::vector<std::pair<int, int>>>, std::vector<NetworKit::node>>
+    BretscherCorneilHabibPaulCographRecognition::
+    LexBfsMinus(bool is_complement, std::vector<NetworKit::node> &a) {
+        std::vector<NetworKit::node> ans(a.size(), INT_MAX);
```

Comment:
Why don't use `reserve()` here + `push_back(x)` instead of `ans[i++] = x` below?

Reply:
fixed